client/my-sites/themes/themes-banner/index.jsx

```diff
@@ -0,0 +1,36 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import cssSafeUrl from 'lib/css-safe-url';
+import Button from 'components/button';
+
+export class ThemesBanner extends PureComponent {
+	static propTypes = {
+		title: PropTypes.string.isRequired,
+		description: PropTypes.string.isRequired,
+		action: PropTypes.func,
+		actionLabel: PropTypes.string.isRequired,
+		backgroundImage: PropTypes.string,
+		href: PropTypes.string,
+	};
+
+	render() {
+		const { title, description, actionLabel, action, backgroundImage, href } = this.props;
+		const backgroundStyle = backgroundImage
+			? { backgroundImage: `url( ${ cssSafeUrl( backgroundImage ) } )` }
+			: null;
+		return (
+			<div className="themes-banner" style={ backgroundStyle }>
+				<h1>{ title }</h1>
+				<p>{ description }</p>
+				<Button compact={ true } primary={ !! backgroundImage } onClick={ action } href={ href }>
```

Comment:
Nitpick -- you can simply write `compact` instead of `compact={ true }` -- it's sort of the preferred style with JSX. Not a biggie tho 😄

Reply:
Oh, and out of curiosity -- is it really the presence of a custom `backgroundImage` that should determine whether this is a  `primary` button or not? I found that a bit surprising.

Reply:
If the banner doesn't have a background image set, then the button, if primary, will be blue, which will be blue on blue, which would not provide enough contrast.

Making the button not primary, will make it render as white, which will provide the proper contrast against the default background:

![screen shot 2018-01-09 at 10 41 08 am](https://user-images.githubusercontent.com/5881557/34726211-c12638f0-f529-11e7-994b-c6197db5c4dd.png)

Reply:
That makes sense from a styling POV, but in terms of UX semantics it doesn't seem quite right 😁 

Even on a blue background, the button should be primary no less; so maybe that's something to either address at a styling level, or to reconsider the default background color in order to be able to keep the primary button, and to keep its appearance (blue) consistent with elsewhere in Calypso 😄